src/openrct2/ride/Ride.cpp

```diff
@@ -2555,8 +2553,14 @@ static void ride_mechanic_status_update(sint32 rideIndex, sint32 mechanicStatus)
     rct_peep *mechanic;
 
     ride = get_ride(rideIndex);
-    switch (mechanicStatus) {
-    case RIDE_MECHANIC_STATUS_UNDEFINED:
+
```

Comment:
I don't completely understand the changes in this area. Could you explain them a bit?

Reply:
I can try. All the changes are related through, so I've got to explain the lot.
Previously, the ride_prepare_breakdown() function sets the mechanicStatus (via ride->mechanic_status) to RIDE_MECHANIC_STATUS_UNDEFINED and updates the ride-> lifecycle_flags: the RIDE_LIFECYCLE_DUE_INSPECTION flag is cleared and the RIDE_LIFECYCLE_BREAKDOWN_PENDING is set.
Here in ride_mechanic_status_update(), when the mechanicStatus is RIDE_MECHANIC_STATUS_UNDEFINED the pending breakdown is elevated to a proper breakdown (RIDE_LIFECYCLE_BROKEN_DOWN is set).
The flag changes made in ride_prepare_breakdown are checked in peep_update_heading_to_inspect().  If this happens when the mechanic is in the ride exit, he falls through the floor.

Logically I see no need for ride_prepare_breakdown() to change the RIDE_LIFECYCLE_DUE_INSPECTION flag or to change ride->mechanic_status. A breakdown being prepared is separate from these things.
The code in question for escalating the pending breakdown to a proper breakdown uses this change to ride->mechanic_status as the trigger however. So leaving the ride->mechanic_status as is requires modifying this trigger. The updated conditions for this trigger are that the ride->mechanic_status is one of the three listed: RIDE_MECHANIC_STATUS_UNDEFINED to cover the original case and the other two to cover the modified behaviour (although I'm not certain that RIDE_MECHANIC_STATUS_CALLING is possible ... need to check that) additionally for these two we need to check that there is a breakdown pending but the ride is not yet broken down.

One subtle point here:
I left setting the ride->mechanic_status = RIDE_MECHANIC_STATUS_CALLING only if is currently RIDE_MECHANIC_STATUS_UNDEFINED - this means a mechanic on the way to inspect a ride won't be called when it does breakdown. On the other hand, the ride would then call the nearest mechanic who need not be the same one. Thoughts about this?

Another point that I didn't notice before now:
RIDE_MECHANIC_STATUS_UNDEFINED is not actually set anywhere anymore. I need to add this to peep_update_fixing_sub_state_13().